### PR TITLE
[clang-format] Fix bad indentation with attribute and templated type

### LINKED ISSUE #76314: [clang-format] Attributes forces any variable with templated type to indent incorrectly
The code:
```cpp
[[maybe_unused]]
int a;
[[maybe_unused]]
foo<int> b;
```

is formatted to:
```cpp
[[maybe_unused]]
int a;
[[maybe_unused]]
foo<int>
    b;
```
any variable template is indented like this, making it inconsistent with non-templated variable definitions.

### PR DESCRIPTION
**Pull request by XDeme1**
Fixes llvm/llvm-project#76314

### CODE REVIEW

**Review by HazardyKnusperkeks — APPROVED** [review]

**Comment by XDeme** [comment]
Hi, Thanks for reviewing, could you merge this for me? I don't have write access

File: clang/unittests/Format/FormatTest.cpp [hunk 1]
```diff
@@ -26492,6 +26492,10 @@ TEST_F(FormatTest, BreakAfterAttributes) {
   verifyFormat("[[nodiscard]]\n"
                "Foo& operator-(Foo&);",
                Style);
+
+  verifyFormat("[[maybe_unused]]\n"
+               "foo<int> f;",
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:26497]:
```suggestion
               "foo<int> k;",
```
See below.

File: clang/unittests/Format/FormatTest.cpp [hunk 2]
```diff
@@ -26492,6 +26492,10 @@ TEST_F(FormatTest, BreakAfterAttributes) {
...
+               Style);
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:26498]: Please move it up and insert it into `Code`:
```
  constexpr StringRef Code("[[maybe_unused]] const int i;\n"
                           "[[foo([[]])]] [[maybe_unused]]\n"
                           "int j;\n"
                           "[[maybe_unused]]\n"
                           "foo<int> k;\n"
                           "[[nodiscard]] inline int f(int &i);\n"
```
and update the test cases for `ABS_Never` and `ABS_Always`.

**Review by owenca — APPROVED** [review]

**Comment by HazardyKnusperkeks** [comment]
> Hi, Thanks for reviewing, could you merge this for me? I don't have write access

I could. But I'm no fan of submitting just because one person approved without others having the chance to say something. (As has happened here.)

**Pull request merged**